kernel/tests/read.rs

```diff
@@ -514,6 +518,7 @@ fn read_with_scan_data(
     Ok(())
 }
 
+// TODO: Add some tests that read a table with no stats, to exercise parquet row group skipping.
```

Comment:
We could also read a multi-part checkpoint file and at least verify row group skipping for P&M actions.

Reply:
Seems reasonable, yes. I would assume `with_row_groups` is (a) well-tested on the parquet side and (b) really obvious is we manage to somehow wire it up wrong.

Reply:
TODO complete! It verifies that predicates actually reach the parquet reader from kernel.